pkg/services/sqlstore/preferences.go

```diff
@@ -13,6 +13,50 @@ func (ss *SQLStore) addPreferencesQueryAndCommandHandlers() {
 	bus.AddHandler("sql", ss.GetPreferences)
 	bus.AddHandler("sql", ss.GetPreferencesWithDefaults)
 	bus.AddHandler("sql", ss.SavePreferences)
+	bus.AddHandler("sql", ss.GetPreferencesJsonData)
+	bus.AddHandler("sql", ss.GetPreferencesJsonDataWithDefaults)
+	bus.AddHandler("sql", ss.SavePreferencesJsonData)
+}
+
+func (ss *SQLStore) GetPreferencesJsonDataWithDefaults(ctx context.Context, query *models.GetPreferencesJsonDataWithDefaultsQuery) error {
+	return ss.WithDbSession(ctx, func(dbSession *DBSession) error {
+		params := make([]interface{}, 0)
+		filter := ""
+
+		if len(query.User.Teams) > 0 {
+			filter = "(org_id=? AND team_id IN (?" + strings.Repeat(",?", len(query.User.Teams)-1) + ")) OR "
+			params = append(params, query.User.OrgId)
+			for _, v := range query.User.Teams {
+				params = append(params, v)
+			}
+		}
+
+		filter += "(org_id=? AND user_id=? AND team_id=0) OR (org_id=? AND team_id=0 AND user_id=0)"
```

Comment:
The `org_id` filter is part of every OR-clause and could be optimized. Would suggest adding that as first filter. To optimize further add a check if `UserId` is > 0 and add filter in similar was as for teams. User id could be 0 for anonymous access or use of an API key.